shared/oae/api/oae.api.widget.js

```diff
@@ -275,6 +278,11 @@ define(['exports', 'jquery', 'underscore', 'oae.api.config', 'oae.api.i18n', 'oa
             // The data-widget attribute is removed, to avoid the widget being rendered again
             $element.removeAttr('data-widget');
 
+            // Apply the widget data from the widget loading element on to the widget data for this
+            // widget
+            widgetData[widgetId] = widgetData[widgetId] || {};
+            $.extend(widgetData[widgetId], $element.data());
```

Comment:
Widgets get initiated through the oae.api.widgets.loadWidgets function, which is automatically run once when the page is loaded. In that call, you can pass in the widgetdata. When the initial widget load puts a new widget container in place, loadWidgets will need to be called explicitly and the widget data can be passed in. So it's possible to pass part of the parameters in through the DOM and part through the loadWidgets call.